subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/report/DefaultReportGenerator.java

```diff
@@ -48,21 +52,24 @@ protected void checkResult(PerformanceFlakinessDataProvider flakinessDataProvide
             .forEach(scenario -> {
                 if (scenario.getRawData().stream().allMatch(ScenarioBuildResultData::isBuildFailed)) {
                     buildFailure.getAndIncrement();
-                } else if (isStableScenario(flakinessDataProvider, scenario.getScenarioName())) {
-                    if (scenario.getRawData().stream().allMatch(ScenarioBuildResultData::isRegressed)) {
+                } else if (scenario.getRawData().stream().allMatch(ScenarioBuildResultData::isRegressed)) {
+                    Set<PerformanceFlakinessDataProvider.ScenarioRegressionResult> regressionResults = scenario.getRawData().stream()
+                        .map(rawScenario -> flakinessDataProvider.getScenarioRegressionResult(rawScenario.getScenarioName(), rawScenario.getDifferencePercentage()))
```

Comment:
Can we make this method `getScenarioRegressionResult(BuildSecenarioResult scenario)`? In this way we can make this line shorter.

Reply:
Done.